src/main/Config.h

```diff
@@ -258,6 +258,27 @@ class Config : public std::enable_shared_from_this<Config>
     // configuration) to delay emitting metadata by one ledger.
     bool EXPERIMENTAL_PRECAUTION_DELAY_META;
 
+    // A config parameter that when set uses the BucketList as the primary
+    // key-value store for LedgerEntry lookups
+    bool EXPERIMENTAL_BUCKET_KV_STORE;
+
+    // A config parameter that when set lazily initializes bucket indexes when
+    // needed. This lowers the bucket index memory usage, but significantly
+    // reduces lookup speed. Should never be set on a validator node, node will
+    // fall out of sync with network.
+    bool EXPERIMENTAL_BUCKET_KV_STORE_LAZY_INDEX;
```

Comment:
I'm slightly on the fence about this config, as it seems to try to mitigate a problem elsewhere (lack of sufficient resources outside of core). Can we try to resolve the resource problem first? Also, do you have a rough idea of how much we're actually saving? From what I understand, the reason this works is because in a way we spread out index generation, instead of kicking off a bunch of indexes all at once, is that right?

Reply:
It's about 2/3 to 1/2 the memory footprint. The largest issue with non-lazy indexing is the future buckets that are not part of the bucketlist yet. Once a future bucket is done merging, they index themselves on the background thread so they can be immediately searched when they are eventually added to the bucketlist. This means that, especially for the larger buckets at the lower levels, we have large indexes sitting around and not being used. The memory gains from this flag are primarily from not indexing future buckets.

Reply:
I'm not sure how necessary this config is to parallel catchup, but it was fairly straight forward to implement and helpful in bug finding. We'd have to ping someone more familiar with supercluster to see how necessary this memory reduction really is (or isn't).

Reply:
Looks like this flag was removed, so I'm resolving this.